rasa/core/policies/ensemble.py

```diff
@@ -193,7 +171,7 @@ def _emit_rule_policy_warning(
 
     def train(
         self,
-        training_trackers: List[DialogueStateTracker],
+        training_trackers: List[TrackerWithCachedStates],
```

Comment:
this causes a type clash in line 187 of `ensemble.py` as `trackers_for_policy` returns things of type `DialogueStateTracker`

Reply:
exactly! that's why I returned `Union` in `trackers_for_policy`, but you told me to remove it

Reply:
uargh. Let's readd it then 😬

Reply:
🙈